src/hera/_cli/generate/util.py

```diff
@@ -0,0 +1,49 @@
+import os
+from fnmatch import fnmatch
+from pathlib import Path
+from typing import Generator, Iterable, List, Set
+
+YAML_EXTENSIONS = {".yml", ".yaml"}
+
+
+def expand_paths(source: Path, suffixes: Set[str], recursive: bool = False) -> Generator[Path, None, None]:
```

Comment:
Can we simplify this with https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob or https://docs.python.org/3/library/pathlib.html#pathlib.Path.walk ?

Reply:
Took a look at this but I think `os.walk` is the easiest to follow as `Path.walk` is 3.12+ only and trying to mix the `glob` with the recursive flag and suffixes is horrendous 🫣